test/unit/connection_string.test.ts

```diff
@@ -883,4 +883,56 @@ describe('Connection String', function () {
       });
     });
   });
+
+  describe('non-genuine hosts', () => {
+    beforeEach(() => {
+      process.env.MONGODB_LOG_CLIENT = 'info';
+    });
+
```

Comment:
Had to branch off of NODE-5452 in order to have client component, will rebase once that PR merges.